### PR TITLE
*: fix issue of configure kv timeout not work when disable batch client

### LINKED ISSUE #981: configure kv timeout not work when disable batch client
When disable batch client, the configure kv timeout will be not work.

```
tikv-client.max-batch-size: 0
```

### PR DESCRIPTION
**Pull request by crazycs520**
close #981

### CODE REVIEW

File: internal/locate/region_request.go [hunk 1]
```diff
@@ -1776,7 +1776,7 @@ func (s *RegionRequestSender) onSendFail(bo *retry.Backoffer, ctx *RPCContext, r
 		return errors.WithStack(err)
 	} else if LoadShuttingDown() > 0 {
 		return errors.WithStack(tikverr.ErrTiDBShuttingDown)
-	} else if errors.Cause(err) == context.DeadlineExceeded && req.MaxExecutionDurationMs < uint64(client.ReadTimeoutShort.Milliseconds()) {
+	} else if (errors.Cause(err) == context.DeadlineExceeded || strings.Contains(err.Error(), "context deadline exceeded")) && req.MaxExecutionDurationMs < uint64(client.ReadTimeoutShort.Milliseconds()) {
```
Comment (cfzjywxk) [inline internal/locate/region_request.go:1779]: Need to add explanations here as the conditions are complex, it's better to abstract a function for kv timeout related check.

Comment (crazycs520) [reply internal/locate/region_request.go]: done.

**Review by cfzjywxk — APPROVED** [review]

**Review by ekexium — APPROVED** [review]

**Review by you06 — APPROVED** [review]
Please add some description.

**Comment by cfzjywxk** [comment]
@crazycs520 Need to pick to 6.5 and standalone branch.

**Pull request merged**